aten/src/ATen/native/LinearAlgebraUtils.h

```diff
@@ -514,4 +514,18 @@ static inline void checkLinalgCompatibleDtype(const std::string& fn_name, Scalar
       out_name, " with dtype ", out_type);
 }
 
+/*
+  Two types of 'other' tensors are supported when solving
+  a system of linear equations matmul(input, x) = other:
+  * 1-dimensional (1D) tensor or batch of 1D tensors (vector case)
+  * 2-dimensional (2D) tensor or batch of 2D tensors (matrix case)
+  original torch.solve supported only the matrix case, while NumPy works for both cases
+  for the batched input we need to be able to distinguish them
+*/
+static inline bool linalg_solve_is_vector_rhs(const Tensor& input, const Tensor& other) {
+  auto expected_batched_rhs_shape = IntArrayRef(input.sizes().data(), input.dim() - 1); // input.shape[:-1]
+  bool vector_case = other.dim() == 1 || (input.dim() - 1 == other.dim() && other.sizes().equals(expected_batched_rhs_shape));
+  return vector_case;
+}
+
```

Comment:
Let's probably add a comment why we extract `input.shape[:-1]`, i.e. it is because we expect `input.shape = (..., m, *)` while `other.shape = (..., m)` for the vector case.
Also, it is not clear how broadcasting is handled from the comment, i.e., it looks like the vector case is only possible once certain rules are satisfied which ignore broadcasting. Another words, broadcasting is not considered and maybe it is worth to reflect that in the name of the function? Or probably make it private?

Reply:
Broadcasting is not considered at all. Currently, we have two rules for deciding whether we have the vector case: for Ax=b vector_case = true if b.ndim == 1 or ((A.ndim-b.ndim == 1) and (A.shape[:-1] == b.shape)). The problem of broadcasting the vector case is discussed in this issue https://github.com/pytorch/pytorch/issues/52915.
By making it private are you suggesting putting this function in an anonymous namespace in BatchLinearAlgebra.cpp file?
The matrix case of the right-hand-side(RHS) of the equation is broadcast in torch.solve, torch.linalg.solve, torch.cholesky_solve, torch.triangular_solve using the other function (`_linalg_broadcast_batch_dims`).